argilla/src/argilla/_helpers/_deploy.py

```diff
@@ -0,0 +1,185 @@
+            {"key": "WORKSPACE", "value": "argilla", "description": "The workspace of the space."},
+        ]
+
+        # Check if the space already exists
+        if hf_api.repo_exists(repo_id=repo_id, repo_type="space"):
+            if cls._check_if_stage_can_be_build(hf_api.get_space_runtime(repo_id=repo_id).stage):
+                hf_api.restart_space(repo_id=repo_id)
+
+            if overwrite:
+                for secret in secrets:
+                    hf_api.add_space_secret(
+                        repo_id=repo_id,
+                        key=secret["key"],
+                        value=secret["value"],
+                        description=secret["description"],
+                    )
+
+                if space_hardware:
+                    hf_api.request_space_hardware(repo_id=repo_id, hardware=space_hardware)
+
+                if space_storage:
+                    hf_api.request_space_storage(repo_id=repo_id, storage=space_storage)
+                else:
+                    cls._space_storage_warning()
+        else:
+            if space_storage is None:
+                cls._space_storage_warning()
+
+            hf_api.duplicate_space(
+                from_id=_ARGILLA_SPACE_TEMPLATE_REPO,
+                to_id=repo_id,
+                private=private,
+                exist_ok=True,
+                hardware=space_hardware,
+                storage=space_storage,
+                secrets=secrets,
+            )
+
+        repo_url: RepoUrl = hf_api.create_repo(repo_id=repo_id, repo_type="space", exist_ok=True, space_sdk="docker")
+        api_url: str = (
+            f"https://{cls._sanitize_url_component(org_name)}-{cls._sanitize_url_component(repo_name)}.hf.space/"
+        )
+        cls._log_message(cls, message=f"Argilla is being deployed at: {repo_url}")
+        while cls._check_if_running(hf_api.get_space_runtime(repo_id=repo_id).stage):
+            time.sleep(_SLEEP_TIME)
+            cls._log_message(cls, message=f"Deployment in progress. Waiting {_SLEEP_TIME} seconds.")
+
+        headers = {}
+        if private:
+            headers["Authorization"] = f"Bearer {hf_token}"
+
+        return cls(api_url=api_url, api_key=api_key, headers=headers)
+
+    @staticmethod
+    def _space_storage_warning() -> None:
+        warnings.warn(
+            "No storage provided. The space will not have persistant storage so every 48 hours your data will be reset."
+        )
+
+    @classmethod
+    def _acquire_hf_token(cls, ht_token: Union[str, None]) -> str:
+        """Obtain the Hugging Face authentication token to deploy a space and authenticate."""
+        if ht_token is None:
+            ht_token = get_token()
+        if ht_token is None:
+            if cls._is_interactive():
+                notebook_login()
+            else:
+                login()
+            ht_token = get_token()
+        return ht_token
+
+    @classmethod
+    def _check_if_running(cls, stage: SpaceStage) -> bool:
```

Comment:
I thought this method returned True if the space is running. Maybe we can change the name
```suggestion
    def _is_building(cls, stage: SpaceStage) -> bool:
```